src/op_mode/ssh.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+#
+# Copyright 2017-2023 VyOS maintainers and contributors <maintainers@vyos.io>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import sys
+import glob
+import vyos.opmode
+from vyos.utils.process import cmd
+from vyos.configquery import ConfigTreeQuery
+
+def show_fingerprints(raw: bool, ascii: bool):
+    config = ConfigTreeQuery()
+    if not config.exists("service ssh"):
+        if raw:
+            return {"response": "SSH server is not enabled."}
+        else:
+            return "SSH server is not enabled."
+    else:
+        # Get list of server public keys
+        publickeys = glob.glob("/etc/ssh/*.pub")
+
+        if publickeys:
+            keys = []
+            for keyfile in publickeys:
+                if ascii:
+                    try:
+                        keys.append(cmd("ssh-keygen -l -v -E sha256 -f " + keyfile))
```

Comment:
While `ssh-keygen` doesn't support any machine-readable output (sadly), I believe we can and should make an attempt to return _useful_ ouput in the `raw` case.

For example, suppose we have this:

```
256 SHA256:SCthwOukM2sh24XSgVbz8J+9i1DtEIV4czP7ECVHjtA no comment (ECDSA)
+---[ECDSA 256]---+
| ..   ..ooo+     |
|  .= . +oE=      |
| ...B o.o.=.     |
|..+. = ooo       |
|.= o. +oS.o      |
|*.+ ...oo. .     |
|.B.. .   ..      |
|o..   . ..       |
|.      . ..      |
+----[SHA256]-----+

```
For a user calling this from a script, something like this would be so much easier to work with:

```
{
  "key_size: 256,
  "type": "ECDSA",
  "comment": "no comment",
  "fingerprint": "SCthwOukM2sh24XSgVbz8J+9i1DtEIV4czP7ECVHjtA",
  "ascii_art": ...
}
```

(Since `ssh-keygen` says "no comment" when comment is not set and doesn't signal the _absense_ of a comment, I think we should just always include what it says than try to infer if it's not a key with comment explicitly set to "no comment" somehow).

Reply:
I've made the requested changes. In the process I also "tabulated" the output.

```
SSH server public key fingerprints:

Type       Key Size  Fingerprint                                         Comment
-------  ----------  --------------------------------------------------  -----------
RSA            3072  SHA256:QUsk927nC0ds8AF/CvxTUtBvNDcyEJjoHGQOE+yFZ8U  root@debian
DSA            1024  SHA256:fWZsMLLdPOJ8RoO+Qo8ThizDUlIoDWkhhTaBK28Nlcs  root@vyos
ED25519         256  SHA256:NGYZ85iN4dbzQVWPrzzxpvxlQLo5sndUsjLOYGYISHU  root@debian
ECDSA           256  SHA256:TsFZT0f4TvlJjKjiIcCRTKPOf459v3oXA0IZiPe76jI  root@debian


show_fingerprints --raw:
[
    {
        "type": "RSA",
        "key_size": "3072",
        "fingerprint": "SHA256:QUsk927nC0ds8AF/CvxTUtBvNDcyEJjoHGQOE+yFZ8U",
        "comment": "root@debian"
    },
    {
        "type": "DSA",
        "key_size": "1024",
        "fingerprint": "SHA256:fWZsMLLdPOJ8RoO+Qo8ThizDUlIoDWkhhTaBK28Nlcs",
        "comment": "root@vyos"
    },
    {
        "type": "ED25519",
        "key_size": "256",
        "fingerprint": "SHA256:NGYZ85iN4dbzQVWPrzzxpvxlQLo5sndUsjLOYGYISHU",
        "comment": "root@debian"
    },
    {
        "type": "ECDSA",
        "key_size": "256",
        "fingerprint": "SHA256:TsFZT0f4TvlJjKjiIcCRTKPOf459v3oXA0IZiPe76jI",
        "comment": "root@debian"
    }
]

SSH server public key fingerprints:

Type       Key Size  Fingerprint                                         Comment      ASCII Art
-------  ----------  --------------------------------------------------  -----------  -------------------
RSA            3072  SHA256:QUsk927nC0ds8AF/CvxTUtBvNDcyEJjoHGQOE+yFZ8U  root@debian  +---[RSA 3072]----+
                                                                                      |    .+=+O.+o+o.  |
                                                                                      |     o=@ E o +.oo|
                                                                                      |    . *.+ = + =o+|
                                                                                      |     . o o * *  o|
                                                                                      |        S o X  . |
                                                                                      |         . = .   |
                                                                                      |          . o    |
                                                                                      |           o .   |
                                                                                      |            .    |
                                                                                      +----[SHA256]-----+
DSA            1024  SHA256:fWZsMLLdPOJ8RoO+Qo8ThizDUlIoDWkhhTaBK28Nlcs  root@vyos    +---[DSA 1024]----+
                                                                                      |+O+. .           |
                                                                                      |*+o +            |
                                                                                      |oo.+ .  . o      |
                                                                                      |o o E    = B     |
                                                                                      |.. * . .S = %    |
                                                                                      |  + = o ++ B o   |
                                                                                      | . . o o ++ o    |
                                                                                      |        + .+     |
                                                                                      |         o.      |
                                                                                      +----[SHA256]-----+
...


show_fingerprints --ascii --raw
[
    {
        "type": "RSA",
        "key_size": "3072",
        "fingerprint": "SHA256:QUsk927nC0ds8AF/CvxTUtBvNDcyEJjoHGQOE+yFZ8U",
        "comment": "root@debian",
        "ascii_art": "+---[RSA 3072]----+\n|    .+=+O.+o+o.  |\n|     o=@ E o +.oo|\n|    . *.+ = + =o+|\n|     . o o * *  o|\n|        S o X  . |\n|  
       . = .   |\n|          . o    |\n|           o .   |\n|            .    |\n+----[SHA256]-----+"
    },
    {
        "type": "DSA",
        "key_size": "1024",
        "fingerprint": "SHA256:fWZsMLLdPOJ8RoO+Qo8ThizDUlIoDWkhhTaBK28Nlcs",
        "comment": "root@vyos",
        "ascii_art": "+---[DSA 1024]----+\n|+O+. .           |\n|*+o +            |\n|oo.+ .  . o      |\n|o o E    = B     |\n|.. * . .S = %    |\n|  
+ = o ++ B o   |\n| . . o o ++ o    |\n|        + .+     |\n|         o.      |\n+----[SHA256]-----+"
    },
    {
        "type": "ED25519",
        "key_size": "256",
        "fingerprint": "SHA256:NGYZ85iN4dbzQVWPrzzxpvxlQLo5sndUsjLOYGYISHU",
        "comment": "root@debian",
        "ascii_art": "+--[ED25519 256]--+\n|     .. E   .....|\n|    .  o % .   ..|\n|   . .  % = . o .|\n|    . .= . o +...|\n|       .S.  o o+.|\n|  
      . = o+o= |\n|         +.++++ =|\n|           o+o.=.|\n|          .. .o..|\n+----[SHA256]-----+"
    },
    {
        "type": "ECDSA",
        "key_size": "256",
        "fingerprint": "SHA256:TsFZT0f4TvlJjKjiIcCRTKPOf459v3oXA0IZiPe76jI",
        "comment": "root@debian",
        "ascii_art": "+---[ECDSA 256]---+\n|  ooo ..o . .oo  |\n|  .=.o + o o..   |\n| .. o o +   o.o. |\n|o  o   o o . .+o |\n| o  .   S o  o...|\n|  
.  . * . o  ...|\n|   . .o =   o    |\n|   E=  + . .     |\n|   .+=o.+oo      |\n+----[SHA256]-----+"
    }
]
```